lib/optional_dependency_helper.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+def require_optional_dependency(name)
+  gem name # require the gem
```

Comment:
Based on what I'm seeing we also need to call `require "cohere"` afterwards. It looks like if you just call `gem 'cohere-ruby'`, the gem classes haven't been loaded yet:
```
 -> langchain [optional-dependency*]: irb
irb(main):001:0> gem "cohere-ruby"
=> true
irb(main):002:0> ::Cohere::Client
(irb):2:in `<main>': uninitialized constant Cohere (NameError)
	from /Users/andrei/.asdf/installs/ruby/3.0.0/lib/ruby/gems/3.0.0/gems/irb-1.6.4/exe/irb:9:in `<top (required)>'
	from /Users/andrei/.asdf/installs/ruby/3.0.0/bin/irb:23:in `load'
	from /Users/andrei/.asdf/installs/ruby/3.0.0/bin/irb:23:in `<main>'
irb(main):003:0> require "cohere"
=> true
irb(main):004:0> ::Cohere::Client
=> Cohere::Client
irb(main):005:0>
```

Reply:
Yeah. I had the hunch that was needed. Thanks for confirming